zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -3030,8 +3078,16 @@ interface ManipulateListener {
     void run() throws KeeperException, InterruptedException;
   }
 
+  // Add a persist listener on the path.
+  // Throws UnsupportedOperationException if there is already a recursive persist listener on the
+  // path because it will overwrite that recursive persist listener.
   private void addPersistListener(String path, Object listener) {
     ManipulateListener addListeners = () -> {
+      if (_zkPathRecursiveWatcherTrie.hasListenerOnPath(path)) {
```

Comment:
Curious, we already check that in subscribing listeners right?

Reply:
Sorry I didn't quit follow. We check if there is a  persistent recursive listener on the path when 
1. unsubscribe persistent recursive listener
2. here 
Did I miss anything?